planner/core/logical_plans.go

```diff
@@ -1236,6 +1236,8 @@ type DataSource struct {
 	// colsRequiringFullLen is the columns that must be fetched with full length.
 	// It is used to decide whether single scan is enough when reading from an index.
 	colsRequiringFullLen []*expression.Column
+
+	accessPathMinSelectivity float64
```

Comment:
Better to add some comments here since `DataSource` is a large structure with lots of fields...

Reply:
Updated~